app/views/runs/_split_table.slim

```diff
@@ -1,62 +1,36 @@
-.card
+.card.shadow
   .card-header.text-white Segments
-  table.card-body.table.table-striped.table-hover.mb-0
-    thead
-      tr
-        th.align-right #
-        th.align-left Name
-        - if @run.completed?(Run::REAL) && @run.completed?(Run::GAME)
-          th.align-right Realtime Duration
-          th.align-right Gametime Duration
-          th.align-right Realtime Finished At
-          th.align-right Gametime Finished At
-        - else
+  .table-responsive
+    table.card-body.table.table-striped.table-hover.mb-0
+      thead
+        tr
+          th.align-right #
+          th.align-left Name
           th.align-right Duration
           th.align-right Finished At
-        th.align-right History
-    tbody
-      - run.segments.each_with_index do |segment, index|
-        tr
-          td.align-right= index + 1
-          td.align-left= segment.name.presence || '-'
-          - if @run.completed?(Run::REAL) && @run.completed?(Run::GAME)
-            td.align-right
-              - if segment.gold?(Run::REAL)
-                .gold-split title='Gold'
-              - if segment.skipped?(Run::REAL) || segment.reduced?(Run::REAL)
-                span.time -
-              - else
-                span = format_milliseconds(segment.duration_ms(Run::REAL))
-            td.align-right
-              - if segment.gold?(Run::GAME)
-                .gold-split title='Gold'
-              - if segment.skipped?(Run::GAME) || segment.reduced?(Run::GAME)
-                span.time -
-              - else
-                span = format_milliseconds(segment.duration_ms(Run::GAME))
-            - if segment.skipped?(Run::REAL)
-              td.align-right.time -
-            - else
-              td.align-right = format_milliseconds(segment.end_ms(Run::REAL))
-            - if segment.skipped?(Run::GAME)
-              td.align-right.time -
-            - else
-              td.align-right = format_milliseconds(segment.end_ms(Run::GAME))
-          - else
-            td.align-right
+          - if run.segments.any? { |segment| segment.histories.any? }
```

Comment:
If no splits have a history, this becomes N queries.  Using `includes(:histories)` solves this, while also caching the result for the iteration just below.